plugins/modules/ec2_transit_gateway.py

```diff
@@ -376,36 +346,34 @@ def create_tgw(self, description):
         options["DnsSupport"] = self.enable_option_flag(self._module.params.get("dns_support"))
         options["MulticastSupport"] = self.enable_option_flag(self._module.params.get("multicast_support"))
 
+        params = {"Description": description, "Options": options}
+
         try:
-            response = self._connection.create_transit_gateway(Description=description, Options=options)
-        except (ClientError, BotoCoreError) as e:
-            self._module.fail_json_aws(e)
+            response = create_ec2_transit_gateway(self._connection, **params)
+        except AnsibleEC2Error as e:
```

Comment:
The error will be handled by https://github.com/ansible-collections/amazon.aws/pull/2325/files#diff-11d6a927433f45aa84de2f54fa0adf33e8094ebb417c75ea476ffe61b4e4f587R1806